src/main/java/sk/uniba/fmph/dcs/game_board/AllPlayersTakeReward.java

```diff
@@ -0,0 +1,60 @@
+package sk.uniba.fmph.dcs.game_board;
+
+import sk.uniba.fmph.dcs.stone_age.ActionResult;
+import sk.uniba.fmph.dcs.stone_age.Effect;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AllPlayersTakeReward implements EvaluateCivilisationCardImmediateEffect {
+    private final RewardMenu menu;
+    private boolean used = false;
+    private static final int WOODNUMBER = 1;
+    private static final int CLAYNUMBER = 2;
+    private static final int STONENUMBER = 3;
+    private static final int GOLDNUMBER = 4;
+    private static final int TOOLNUMBER = 5;
+    private static final int FIELDNUMBER = 6;
+    private ArrayList<Integer> diceThrows;  //used for tests, null if throws should be random
+
+    public AllPlayersTakeReward(final RewardMenu menu, final ArrayList<Integer> diceThrows) {
+        this.menu = menu;
+        this.diceThrows = diceThrows;
+    }
+
+    private static Map<Integer, Effect> resourceMapInitiate() {
+        Map<Integer, Effect> map = new HashMap<>();
+        map.put(WOODNUMBER, Effect.WOOD);
+        map.put(CLAYNUMBER, Effect.CLAY);
+        map.put(STONENUMBER, Effect.STONE);
+        map.put(GOLDNUMBER, Effect.GOLD);
+        map.put(TOOLNUMBER, Effect.TOOL);
+        map.put(FIELDNUMBER, Effect.FIELD);
+        return map;
+    }
+
+    private ArrayList<Effect> menuItemsInitiate(final Player player) {
+        int numberOfPlayers = player.playerOrder().getPlayers();
+        if (diceThrows == null) {
+            Throw t = new Throw();
+            diceThrows = t.throwDice(numberOfPlayers);

```

Comment:
```suggestion
        if (diceThrows == null) {
            diceThrows = t.throwDice(numberOfPlayers);

```
Bolo by fajn dostat instanciu triedy throw rovno v konstruktore aby sa nevytvarala vzdy nova

Reply:
ale nemusi to tak nutne byt

Reply:
Toto je zle lebo diceThrows == null bude platit len prve zavolanie performEffect. Lepsie bude ak bude mozno nejaky boolean random, ktory rozhodne ci chceme testovaciu verziu alebo normalnu